modules/programs/neovim.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.neovim;
+in
+{
+  options = {
+    programs.neovim = {
+      enable = mkEnableOption "Neovim";
+
+      withPython = mkOption {
+        type = types.nullOr types.bool;
+        default = true;
+        description = ''
+          Enable python2 provider. Set to true to use python2-based plugins
```

Comment:
Instead of "python2" (and "python3" below) I'd suggest to write "Python 2" (and "Python 3"). Looks nicer and seems to be what is used [officially](https://wiki.python.org/moin/Python2orPython3).